src/linux/mod.rs

```diff
@@ -37,31 +34,36 @@ static KVM: LazyLock<Kvm> = LazyLock::new(|| Kvm::new().unwrap());
 /// It is used to stop a vCPU from another thread.
 pub(crate) struct KickSignal;
 
+// TODO: nix::signal::Signal doesn't support real-time signals yet.
+// Start using the Signal type once this no longer is the case.
+//
+// See:
+// - https://github.com/nix-rust/nix/issues/495
+// - https://github.com/nix-rust/nix/issues/1143
+// - https://github.com/hermit-os/uhyve/issues/1083
 impl KickSignal {
 	const RTSIG_OFFSET: libc::c_int = 0;
 
-	fn get() -> Signal {
-		let kick_signal = SIGRTMIN() + Self::RTSIG_OFFSET;
+	fn get() -> libc::c_int {
+		let kick_signal: libc::c_int = SIGRTMIN() + Self::RTSIG_OFFSET;
 		assert!(kick_signal <= SIGRTMAX());
-		// TODO: Remove the transmute once realtime signals are properly supported by nix
-		// https://github.com/nix-rust/nix/issues/495
-		unsafe { std::mem::transmute(kick_signal) }
+		kick_signal
 	}
 
 	pub(crate) fn register_handler() -> nix::Result<()> {
 		extern "C" fn handle_signal(_signal: libc::c_int) {}
-		// SAFETY: We don't use the `signal`'s return value.
-		unsafe {
-			signal(Self::get(), SigHandler::Handler(handle_signal))?;
-		}
-		Ok(())
+		// SAFETY: We don't use the `signal`'s return value and use an empty handler.
+		// (Sidenote: SIG_DFL and SIG_IGN don't do the trick.)
```

Comment:
SIG_DFL terminates the process upon a VM exit (i.e. a hypercall), SIG_IGN prevents shutdowns from taking place (but all other exits continue to work). I left this hint there for the next person to touch this code.